src/Updater.php

```diff
@@ -371,19 +385,41 @@ protected function downloadPhar()
 
     protected function replacePhar()
     {
-        rename($this->getTempPharFile(), $this->getLocalPharFile());
+        $localPharFile = $this->getLocalPharFile();
+        if (!is_writable($localPharFile)) {
```

Comment:
getLocalPharFile() is also called by setTempDirectory(), which would cause this to fail unnecessarily during a check also.  I understand wanting to reduce code duplication, but I think it also makes sense to have these exceptions associated with the code that requires writability.  If you'd really like to avoid duplicating the exceptions, I would propose an interstitial method, like `protected isDirectoryWritable($this->getTempDirectory);` which just does the writability check and throws the exception.  I decided not to do that originally because it actually creates more lines of code, but perhaps you'd prefer that to the current PR?